### PR TITLE
Clean up Timer tests

### PR DESCRIPTION
**Pull request by stephentoub**
- Add argument names to Assert.Throws
- Add coverage for some uncovered cases
- Put test classes in namespaces
- Consolidate some tests

Depends on https://github.com/dotnet/coreclr/pull/25617

(Best reviewed with whitespace ignored: https://github.com/dotnet/corefx/pull/39333/files?w=1)

### CODE REVIEW

File: src/System.Threading.Timer/tests/TimerFiringTests.cs [hunk 1]
```diff
@@ -2,339 +2,361 @@
...
+        [Fact]
+        public void Timer_ChangeToDelete_DoesntFire()
         {
-            if (Interlocked.Decrement(ref callbacks) >= 0)
+            const int DueTime = 1000;
+            var are = new AutoResetEvent(false);
+            using (var t = new Timer(_ => are.Set(), null, DueTime, -1))
             {
-                Assert.True(b.SignalAndWait(MaxPositiveTimeoutInMs));
+                t.Change(-1, -1);
```
Comment (danmosemsft) [inline src/System.Threading.Timer/tests/TimerFiringTests.cs]: Nit, `Timeout.Infinite` is more readable than `-1` (would have saved me looking at MSDN)

File: src/System.Threading.Timer/tests/TimerFiringTests.cs [hunk 2]
```diff
@@ -2,339 +2,361 @@
...
+                Assert.False(are.WaitOne(DueTime * 2));
```
Comment (danmosemsft) [inline src/System.Threading.Timer/tests/TimerFiringTests.cs]: There's a race here right, where the timer might fire before `Change` is called?

File: src/System.Threading.Timer/tests/TimerChangeTests.cs [hunk 3]
```diff
@@ -2,115 +2,82 @@
...
-
-    [Fact]
-    public void Timer_Change_Int64_Negative()
-    {
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Timer(EmptyTimerTarget).Change((long)-2, (long)-1));
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Timer(EmptyTimerTarget).Change((long)-1, (long)-2));
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Timer(EmptyTimerTarget).Change((long)0xffffffff, (long)-1));
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Timer(EmptyTimerTarget).Change((long)-1, (long)0xffffffff));
-    }
+            var are = new AutoResetEvent(false);
```
Comment (danmosemsft) [inline src/System.Threading.Timer/tests/TimerChangeTests.cs]: Dispose AutoResetEvent ?

File: src/System.Threading.Timer/tests/TimerFiringTests.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Threading.Timer/tests/TimerFiringTests.cs]: There is. My hope is that I picked a duration long enough that will never get false positives. If I'm wrong, we can add a retry. Couldn't think of a better way to test this unfortunately.

File: src/System.Threading.Timer/tests/TimerChangeTests.cs (hunk 3 above)
Comment (stephentoub) [reply src/System.Threading.Timer/tests/TimerChangeTests.cs]: The problem with that in these async tests is that if someone goes wrong and the timer callback end up executing after the are is disposed, we'll get disposed exceptions. It's probably not an issue here specifically, but that's why the rest of the tests leave it for finalization, and got consistency I've done the same.

File: src/System.Threading.Timer/tests/TimerFiringTests.cs (hunk 2 above)
Comment (danmosemsft) [reply src/System.Threading.Timer/tests/TimerFiringTests.cs]: Perhaps you could use a longer duration, and then await a sleep, then check whether the timer fired or not?  Then another test could run meantime?
Rarely are outer loop timer tests run locally anyway.
I don't have a strong opinion I've just gotten used to every possible manifestation of slow machines.

**Pull request merged**